misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/CustomSystemClassLoader.java

```diff
@@ -0,0 +1,116 @@
+
+public class CustomSystemClassLoader extends ClassLoader {
+
+    private final HashMap<String, byte[]> modifiedClassDefs;
+    private final List<String> excludingClasses = new ArrayList<>();
+
+    public CustomSystemClassLoader() {
+        super(getSystemClassLoader());
+        this.modifiedClassDefs = new HashMap<>();
+
+        populateExcludingClasses();
+    }
+
+    private void populateExcludingClasses() {
+        try(InputStream is = BTestMain.class.getResourceAsStream("/" + ProjectConstants.EXCLUDING_CLASSES_FILE);
+            BufferedReader br = new BufferedReader(new InputStreamReader(Objects.requireNonNull(is)))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                excludingClasses.add(line);
+            }
+        }
+        catch (NullPointerException | IOException e) {
+            throw new RuntimeException("Error reading excludingClasses.txt", e);
+        }
+
+    }
+
+    public CustomSystemClassLoader(Map<String, byte[]> modifiedClassDefs) {
+        super(getSystemClassLoader());
+        this.modifiedClassDefs = new HashMap<>(modifiedClassDefs);
+
+        populateExcludingClasses();
+    }
+
+    @Override
+    public Class<?> loadClass(String name) throws ClassNotFoundException {
+        Class<?> loadedClass;
+
+        //return the class if it is already loaded
+        loadedClass = findLoadedClass(name);
+        if (loadedClass != null) {
+            return loadedClass;
+        }
+
+        //if the class is an inbuilt class, delegate the classloading to the system classloader
+        if (name.startsWith("java.") || name.startsWith("javax.")) {
+            return super.loadClass(name);
+        }
+
+        //if the class is in not in the excludingClasses list, delegate the classloading to the system classloader
+        if (!excludingClasses.contains(name)) {
+            return super.loadClass(name);
+        }
+
+        return findClass(name);
+    }
+
+    @Override
+    public Class<?> findClass(String name) throws ClassNotFoundException {
+        byte[] classBytes = this.modifiedClassDefs.remove(name);
+        if (classBytes != null) {
+            return defineClass(name, classBytes, 0, classBytes.length);
+        }
+
+        try {
+            classBytes = getClassBytes(name);
+            return defineClass(name, classBytes, 0, classBytes.length);
+        } catch (IOException e) {
+            return super.findClass(name);
+        }
+    }
+
+    private byte[] getClassBytes(String classFileName) throws IOException {
+        String path = File.separator + classFileName.replace('.', File.separatorChar) + ".class";
+        InputStream is = BTestMain.class.getResourceAsStream(path);
+
+        int size = Objects.requireNonNull(is).available();
+
+        byte[] buff = new byte[size];
+
+        DataInputStream dis = new DataInputStream(is);
+        dis.readFully(buff);
+        dis.close();
+
+        return buff;
+    }
+}
```

Comment:
There are redundant new lines used in most classes. You can check and remove them